### PR TITLE
Kraken: take creation of stop_time into account from FIRST

### PR DESCRIPTION
**Pull request by xlqian**
https://jira.kisio.org/browse/NAVP-1047

### CODE REVIEW

File: source/kraken/realtime.cpp [hunk 1]
```diff
@@ -332,14 +339,16 @@ create_disruption(const std::string& id,
...
-                most_important_stoptime_status = std::max(most_important_stoptime_status, arrival_status);
+                most_important_stoptime_status = std::max({most_important_stoptime_status,
+                    departure_status ,arrival_status});
 
                 StopTimeUpdate st_update{std::move(stop_time), message, departure_status, arrival_status};
                 impact->aux_info.stop_times.emplace_back(std::move(st_update));
             }
 
             switch (most_important_stoptime_status) {
+            case StopTimeUpdate::Status::ADDED:
```
Comment (nonifier) [inline source/kraken/realtime.cpp:350]: It would have been nice to extract this in a separate function as it's already super loooooooooong.

Comment (nonifier) [inline source/kraken/realtime.cpp]: It would have been nice to extract this in a separate function as it's already super loooooooooong.

Comment (nonifier) [inline source/kraken/realtime.cpp]: It would have been nice to extract this in a separate function as it's already super loooooooooong.

Comment (nonifier) [inline source/kraken/realtime.cpp]: It would have been nice to extract this in a separate function as it's already super loooooooooong.

**Review by nonifier — APPROVED** [review]

**Comment by xlqian** [comment]
The addition of stop_time works! (see the 6th stop_time)
image

File: source/kraken/realtime.cpp [hunk 2]
```diff
@@ -193,10 +193,17 @@ static nt::disruption::StopTimeUpdate::Status get_status(const transit_realtime:
         return st.schedule_relationship();
     };
 
-    if (get_relationship(event, st) ==
-            transit_realtime::TripUpdate_StopTimeUpdate_ScheduleRelationship_SKIPPED) {
+    auto transit_reltime_status = get_relationship(event, st);
+    switch (transit_reltime_status) {
```
Comment (pbougue) [inline source/kraken/realtime.cpp:197]: typo: transit_re`a`ltime_status

**Review by pbougue — COMMENTED** [review]
Is it possible to add test on that ?

Also, why is there no name for the station on your screenshot ?

**Comment by xlqian** [comment]
this pr is to be merged into https://github.com/CanalTP/navitia/pull/2581

**Pull request closed without merging**